### ISSUE TITLE
[oauth2] crash on set header "x-authenticated-scope" after migrated to 1.0.0rc2

### ISSUE THREAD

**Issue by seamons** [Kong/kong#3964]
### Summary

After migrating kong from version 0.14.1 to 1.0.0rc2 following the instrucstions [here](https://discuss.konghq.com/t/kong-1-0rc2-available-for-testing/2142), every request validated by oauth2 plugins crashes and produce the some error (see detailed logs below).

I did not provide `scope` parameter when provisioning the oauth2 token and the `config.scopes` settings for oauth2 plugin is null.

### Steps To Reproduce

1. Just send request will be validated by the oauth2 plugin after migrating from 0.14.1 to 1.0.0rc2

### Additional Details & Logs

- Kong version (`$ kong version`)
  1.0.0rc2
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
> 2018/11/09 03:24:34 [error] 29#0: *224450 lua coroutine: runtime error: /usr/local/share/lua/5.1/kong/plugins/oauth2/access.lua:570: invalid header value for "x-authenticated-scope": got nil, expected string, number or boolean
 stack traceback:
 coroutine 0:
       [C]: in function 'error'
       /usr/local/share/lua/5.1/kong/pdk/private/checks.lua:57: in function 'validate_header'
       /usr/local/share/lua/5.1/kong/pdk/service/request.lua:260: in function 'set_header'
       /usr/local/share/lua/5.1/kong/plugins/oauth2/access.lua:570: in function 'set_consumer'
       /usr/local/share/lua/5.1/kong/plugins/oauth2/access.lua:626: in function 'do_authentication'
       /usr/local/share/lua/5.1/kong/plugins/oauth2/access.lua:655: in function 'execute'
       /usr/local/share/lua/5.1/kong/plugins/oauth2/handler.lua:12: in function </usr/local/share/lua/5.1/kong/plugins/oauth2/handler.lua:10>
 coroutine 1:
       [C]: in function 'resume'
       coroutine.wrap:21: in function <coroutine.wrap:21>
       /usr/local/share/lua/5.1/kong/init.lua:659: in function 'access'
- Kong configuration
  - oauth2 plugin config
```json
{
    "config": {
        "accept_http_if_already_terminated": true,
        "auth_header_name": "authorization",
        "enable_authorization_code": true,
        "enable_client_credentials": false,
        "enable_implicit_grant": false,
        "enable_password_grant": true,
        "global_credentials": true,
        "hide_credentials": false,
        "mandatory_scope": false,
        "provision_key": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "refresh_token_ttl": 1209600,
        "token_expiration": 604800
    },
    "created_at": 1531216048,
    "enabled": true,
    "id": "xxxxxxxx-05c8-4c5d-a756-86d926b8799a",
    "name": "oauth2",
    "route": {
        "id": "xxxxxxxx-23cf-488b-b74a-e450c054638e"
    }
}
```
- Operating system
docker image: kong:1.0.0rc2

**Comment by bungle** [comment]
Good catch! Thank you!

**Comment by bungle** [comment]
But maybe that is good thing, so we can check was the intended behavior to remove or skip.

**Comment by hishamhm** [comment]
Fixed by #4006. Thanks for reporting!
